src/overlays/actors/ovl_Bg_Mori_Idomizu/z_bg_mori_idomizu.c

```diff
@@ -20,23 +35,156 @@ const ActorInit Bg_Mori_Idomizu_InitVars = {
     (ActorFunc)BgMoriIdomizu_Update,
     NULL,
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Mori_Idomizu/func_808A4640.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Mori_Idomizu/func_808A4648.s")
+void BgMoriIdomizu_SetActionFunction(BgMoriIdomizu* this, BgMoriIdomizuActionFunc actionFunc) {
+    this->actionFunc = actionFunc;
+}
+
+void BgMoriIdomizu_SetWaterLevel(GlobalContext* globalCtx, s16 waterLevel) {
+    WaterBox* waterBox;
+
+    waterBox = globalCtx->colCtx.stat.colHeader->waterBoxes;
+    waterBox[2].unk_02 = waterLevel;
+    waterBox[3].unk_02 = waterLevel;
+    waterBox[4].unk_02 = waterLevel;
+}
+
+void BgMoriIdomizu_Init(Actor* thisx, GlobalContext* globalCtx) {
+    GlobalContext* globalCtx2 = globalCtx;
+    BgMoriIdomizu* this = THIS;
+
+    if (alreadyLoaded != 0) {
+        Actor_Kill(&this->actor);
+        return;
+    }
+    this->actor.scale.x = 1.1f;
+    this->actor.scale.y = 1.0f;
+    this->actor.scale.z = 1.0f;
+    this->actor.posRot.pos.x = 119.0f;
+    this->actor.posRot.pos.z = -1820.0f;
+    this->switchFlag = Flags_GetSwitch(globalCtx2, this->actor.params & 0x3F);
+    if (this->switchFlag != 0) {
```

Comment:
```suggestion
    this->switchFlagSetPrev = Flags_GetSwitch(globalCtx2, this->actor.params & 0x3F);
    if (this->switchFlagSetPrev) {
```
The result indicates if the switch flag is set or cleared, `switchFlag` is a confusing name for that intent
I suggest suffixing `Prev` here instead of prefixing `new` in the main action function